### PR TITLE
[`tests`] add slow tests to GH workflow

### PR DESCRIPTION
**Pull request by younesbelkada**
This PR adds the workflow to run slow tests

### CODE REVIEW

**Comment by younesbelkada** [comment]
Added the corresponding GCP machine to the list of hosted runners, I think we should be all good once we add the `SLACK_API_TOKEN` inside the repo's secret!

**Review by muellerzr — APPROVED** [review]
Looks good to me! I'll note here that occasionally the nightly reporter bugs out, but for now this does the job :) (I'll ping when I figure out a better reporting setup :) )

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      - name: Generate Report
+        if: always()
+        run: |
+          pip install slack_sdk
+          python scripts/log_reports.py >> $GITHUB_STEP_SUMMARY
+
+  run_all_tests_multi_gpu:
+    runs-on: [self-hosted, docker-gpu, multi-gpu]
+    env:
+      CUDA_VISIBLE_DEVICES: "0,1"
```
Comment (ydshieh) [inline .github/workflows/nightly.yml:60]: same

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  RUN_SLOW: "yes"
+  IS_GITHUB_CI: "1"
+  SLACK_API_TOKEN: ${{ secrets.SLACK_API_TOKEN }}
+
+
+jobs:
+  run_all_tests_single_gpu:
+    runs-on: [self-hosted, docker-gpu, multi-gpu]
+    env:
+      CUDA_VISIBLE_DEVICES: "0"
```
[the remark this answers is not in the document]
Comment (muellerzr) [reply .github/workflows/nightly.yml:18]: The issue with matrix (imo) is it's less clear, vs running them this way when we view the reports in GHA. And matrix get messy quick as well :) 

But you are right, both options work. @younesbelkada definitely worth I think playing with both to see which you prefer more eventually :)

[the remark this answers is not in the document]
Comment (muellerzr) [reply .github/workflows/nightly.yml:18]: Though thinking on this more, probably worth doing on our nightlies. Good point @ydshieh. Most of my fear resided in matrix upon matrix, but the nightlies don't have split tests like our main does.

**Review by pacman100 — APPROVED** [review]
Thank you so much @younesbelkada, @muellerzr and @ydshieh for setting up the slow tests and Slack notifications! 🚀🤗✨

**Comment by younesbelkada** [comment]
Thank you all for your comments!

**Pull request merged**